RocketChat/Rocket.Chat
# feat: [ENTERPRISE] Automatic deletion of LDAP synced users already deleted in AD

## Issue Description
Feature that automatically deletes LDAP-synced users that were already deleted in the AD.

## Steps to Reproduce
1. Setup LDAP integration with RC
2. Sync your AD users to RC and confirm they all sync fine.
3. Delete a user in your AD

## Expected Behavior
Since the synced user has been deleted in AD and no longer exists there, the user should automatically be deleted in RC at the next sync job.

## Actual Behavior
The synced user has to be manually deleted in RC. So if you have a lot of users deleted in the AD, you will have to manually delete each of them in RC.

## Test Plan
1. Setup LDAP integration with RC;
2. Turn on Background Sync on the admin panel (Admin -> LDAP -> Enterprise -> Background Sync);
3. On the same pane, turn on the options `Background Sync Import New Users` and `Background Sync Remove Deleted Users`;
4. Sync your AD users to RC and confirm they all sync fine;
5. Delete a user in your AD;
6. Sync the LDAP again.

**IMPORTANT:** in order to avoid mass deletions by mistake, we only delete RC users when at least one user could be found via LDAP.

TC-383
